plugins/survey/src/types.ts

```diff
@@ -0,0 +1,153 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import { AttachedDoc, Class, CollectionSize, Doc, Markup, Ref, Space } from '@hcengineering/core'
+import { Attachment } from '@hcengineering/attachment'
+import { Person } from '@hcengineering/contact'
+import { ComponentType, SvelteComponent } from 'svelte'
+import { Resource } from '@hcengineering/platform'
+
+/**
+ * @public
+ *
+ * @link https://www.npmjs.com/package/lexorank
+ */
+export type Rank = string
+
+/**
+ * ∈ [0, 1]
+ */
+export type Fraction = number
+
+/** @public */
+export interface Survey extends Space {
+  name: string
+  questions: CollectionSize<Question>
+  requests: CollectionSize<Request>
+  // TODO: isAssessment: boolean
+}
+
+/** @public */
+export interface Question extends AttachedDoc<Survey, 'questions', Survey> {
+  title: Markup
+  rank: Rank
+  attachments: CollectionSize<Attachment>
+  assessment?: AssessmentData<this>
+}
+
+/** @public */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export interface AssessmentData<Q extends Question> {
+  weight: number
+}
+
+/** @public */
+// eslint-disable-next-line @typescript-eslint/no-empty-interface, @typescript-eslint/no-unused-vars
+export interface AnswerData<Q extends Question> {}
+
+/** @public */
+export interface QuestionOption {
+  label: Markup
+}
+
+/** @public */
+export interface SingleChoiceQuestion extends Question {
+  options: QuestionOption[]
+  shuffle: boolean
+}
+
+/** @public */
+export interface SingleChoiceAnswerData extends AnswerData<SingleChoiceQuestion> {
+  selection?: number
+}
+
+/** @public */
+export interface SingleChoiceAssessmentData extends AssessmentData<SingleChoiceQuestion> {
+  correctAnswer: SingleChoiceAnswerData
+}
+
+/** @public */
+export interface MultipleChoiceQuestion extends Question {
+  options: QuestionOption[]
+  shuffle: boolean
+}
+
+/** @public */
+export interface MultipleChoiceAnswerData extends AnswerData<MultipleChoiceQuestion> {
+  selections: number[]
+}
+
+/** @public */
+export interface MultipleChoiceAssessmentData extends AssessmentData<MultipleChoiceQuestion> {
+  correctAnswer: MultipleChoiceQuestion
+}
+
+/** @public */
+export interface ReorderQuestion extends Question {
+  options: QuestionOption[]
+  shuffle: boolean
+}
+
+/** @public */
+export interface ReorderAnswerData extends AnswerData<ReorderQuestion> {
+  order: number[]
+}
+
+/** @public */
+export interface ReorderAssessmentData extends AssessmentData<ReorderQuestion> {
+  correctAnswer: ReorderQuestion
+}
+
+/** @public */
+export interface SurveyRequest extends AttachedDoc<Survey, 'requests', Survey> {
+  // TODO: Should it be Employee, no?
+  // TODO: Should it be an array of assignees instead?
```

Comment:
I guess we need one request per surveyed person

Reply:
Yeah, for now that's my thinking, too. Uberflow implementation has it as an array though. I'll leave this TODO until I have a complete UI flow implemented, and see how it plays out.

Reply:
One of the use cases in qms is when a training manager assigns a training to a group of people and wants to track its overall progress e.g. 18/29 passed the training. For this it might be easier to have it as an array.